core/src/engine/io/tls_wrapper_test.cpp

```diff
@@ -139,6 +139,107 @@ constexpr auto kShortTimeout = std::chrono::milliseconds{10};
 
 }  // namespace
 
+UTEST(TlsWrapper, InitListSmall) {
+  const std::string kString(16, 'a');
+  const engine::io::IoData kData{kString.data(), kString.size()};
+  const auto deadline = Deadline::FromDuration(utest::kMaxTestWaitTime);
+
+  TcpListener tcp_listener;
+  auto [server, client] = tcp_listener.MakeSocketPair(deadline);
+
+  auto server_task = utils::Async(
+      "tls-server",
+      [deadline, kData](auto&& server) {
+        auto tls_server = io::TlsWrapper::StartTlsServer(
+            std::forward<decltype(server)>(server),
+            crypto::Certificate::LoadFromString(cert),
+            crypto::PrivateKey::LoadFromString(key), deadline);
+        if (tls_server.WriteAll({kData, kData, kData, kData}, deadline) !=
+            kData.len * 4) {
+          throw std::runtime_error("Couldn't send data");
+        }
+      },
+      std::move(server));
+
+  auto tls_client =
+      io::TlsWrapper::StartTlsClient(std::move(client), {}, deadline);
+  std::vector<char> buffer(kData.len * 4);
+  const auto bytes_rcvd =
+      tls_client.RecvAll(buffer.data(), buffer.size(), deadline);
+
+  server_task.Get();
+  std::string result(buffer.data(), bytes_rcvd);
+  EXPECT_EQ(result, kString + kString + kString + kString);
+}
+
+UTEST(TlsWrapper, InitListLarge) {
+  const std::string kString(8'192, 'a');
+  const engine::io::IoData kData{kString.data(), kString.size()};
+  const auto deadline = Deadline::FromDuration(utest::kMaxTestWaitTime);
+
+  TcpListener tcp_listener;
+  auto [server, client] = tcp_listener.MakeSocketPair(deadline);
+
+  auto server_task = utils::Async(
+      "tls-server",
+      [deadline, kData](auto&& server) {
+        auto tls_server = io::TlsWrapper::StartTlsServer(
+            std::forward<decltype(server)>(server),
+            crypto::Certificate::LoadFromString(cert),
+            crypto::PrivateKey::LoadFromString(key), deadline);
+        if (tls_server.WriteAll({kData, kData, kData, kData}, deadline) !=
```

Comment:
Use different strings, for example
```
const std::string kStringA(16, 'a');
const std::string kStringB(16, 'b');
const std::string kStringC(16, 'c');
const std::string kStringD(16, 'd');
```

In that case after `if (tls_server.WriteAll({kDataA, kDataB, kDataC, kDataD}, deadline) !=` you would be able to make sure that the strings were received in right order `EXPECT_EQ(result, kStringA + kStringB + kStringC + kStringD);`